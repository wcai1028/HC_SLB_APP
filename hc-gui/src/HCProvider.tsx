import React from 'react'
import { Store } from 'redux'
import { A10Provider } from 'a10-gui-framework'

export interface IHCPRoviderProps {
  children: JSX.Element
  store?: Store<any>
  initState?: IObject
}

import appReducers from './redux/reducers'
import config from './settings/config'
const middlewares: any[] = []

const HCProvider = (props: IHCPRoviderProps) => {
  const { children, store, initState } = props
  const options: any = {
    initState,
    reducers: appReducers,
    middlewares,
    CONFIG: config,
    children,
  }
  if (store) {
    options.store = store
  }
  return <A10Provider {...options} />
}
export default HCProvider
