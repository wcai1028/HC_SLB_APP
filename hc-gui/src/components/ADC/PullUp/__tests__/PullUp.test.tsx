import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'
import * as moxios from 'moxios'
import HCProvider from 'src/HCProvider'

const renderTestComponent = () => {
  const PullUp = require('../index').default
  return ReactTestRenderer.create(
    <HCProvider>
      <PullUp />
    </HCProvider>,
  )
}
const renderTestComponentWithDisable = () => {
  const A10IconTextGroup = require('../index').default
  return ReactTestRenderer.create(
    <HCProvider>
      <A10IconTextGroup disabled="true" />
    </HCProvider>,
  )
}

const getComponentInstance = (testRenderer: any) => {
  const testInstance = testRenderer.root
  const testComponent = testInstance.find(
    (node: any) => node.type.name === 'PullUp',
  )
  return testComponent.instance
}
beforeEach(() => {
  moxios.install()
})
afterEach(() => {
  moxios.uninstall()
})

describe('test component: PullUp', () => {
  test('should render properly', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent()
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot()
      done()
    })
  })
})

describe('test function literally', () => {
  test('onDragStart should render properly', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent()
    const instance = getComponentInstance(testComponent)
    instance.onDragStart({ clientX: 250, clientY: 250 })
    setImmediate(() => {
      expect(instance.canMove).toBe(true)
      done()
    })
  })
  test('onDragEnd should render properly', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent()
    const instance = getComponentInstance(testComponent)
    instance.onDragEnd()
    setImmediate(() => {
      expect(instance.canMove).toBe(false)
      done()
    })
  })
  test('onDrag should render properly', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent()
    const instance = getComponentInstance(testComponent)
    const spyOnOnMove = jest.spyOn(instance, 'onMove')
    instance.onDrag({ clientX: 150, clientY: 150 })
    setImmediate(() => {
      expect(spyOnOnMove).toHaveBeenCalledTimes(1)
      spyOnOnMove.mockRestore()
      done()
    })
  })
  test('onMove should render properly', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent()
    const instance = getComponentInstance(testComponent)
    instance.canMove = true
    instance.onMove({ clientX: 200, clientY: 200 })
    setImmediate(() => {
      expect(instance.state.height).toBe(-100)
      done()
    })
  })
})
