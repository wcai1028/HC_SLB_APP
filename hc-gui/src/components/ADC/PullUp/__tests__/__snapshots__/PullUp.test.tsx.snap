// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: PullUp should render properly 1`] = `
<div
  style={
    Object {
      "paddingBottom": "100px",
      "position": "relative",
    }
  }
>
  <div
    className="pullup"
    style={
      Object {
        "height": undefined,
        "minHeight": undefined,
      }
    }
  >
    <div
      className="pullupContainer pullup-container"
    >
      <div
        className="pullupHeader"
      >
        <div
          className="pullupHeaderBar"
          draggable={true}
          onDrag={[Function]}
          onDragEnd={[Function]}
          onDragStart={[Function]}
          title="Pull"
        >
          <div />
          <div />
        </div>
      </div>
      <div
        style={
          Object {
            "height": NaN,
            "overflowY": "auto",
          }
        }
      />
    </div>
  </div>
</div>
`;
